src/prototype_creep_resources.js

```diff
@@ -446,8 +446,8 @@ Creep.prototype.moveToSource = function(source) {
     this.memory.routing = {};
   }
   this.memory.routing.reverse = false;
-  if (this.room.memory.misplacedSpawn || this.room.controller.level < 3) {
-    this.moveToMy(source.pos);
+  if (this.room.memory.misplacedSpawn || this.room.controller.level < 3 || this.memory.role === 'nextroomer') {
```

Comment:
on restoring high level room they are staying in line instead of harvesting from all available spots near source